app/controllers/new-signup.js

```diff
@@ -41,6 +43,35 @@ export default class NewSignUpController extends Controller {
     this.analytics.trackEvent(NEW_SIGNUP_FLOW.USER_GETTING_STARTED);
   }
 
+  async generateUsername(firstname, lastname) {
```

Comment:
minor fix for return statements

```

async generateUsername(firstname, lastname) {
    if (typeof firstname !== 'string' || typeof lastname !== 'string') {
        throw new Error('Invalid input: firstname and lastname must be strings');
    }

    try {
        const sanitizedFirstname = firstname.toLowerCase();
        const sanitizedLastname = lastname.toLowerCase();
        const response = await fetch(
            `${ENV.BASE_API_URL}/users/username?dev=true&firstname=${sanitizedFirstname}&lastname=${sanitizedLastname}`,
            {
                method: 'GET',
                headers: {
                    'Content-Type': 'application/json',
                },
                credentials: 'include',
            }
        );
        const user = await response.json();
        if (user && user.username) {
            return user;
        }
        throw new Error('Username generation failed: Invalid response from server');
    } catch (err) {
        this.toast.error(
            ERROR_MESSAGES.usernameGeneration,
            'error!',
            toastNotificationTimeoutOptions
        );
        throw err; // Re-throw the error after showing the toast
    }
}
````

Reply:
Resolved this in the new commit please check